aten/src/ATen/native/cuda/AmpKernels.cu

```diff
@@ -67,15 +70,103 @@ void _amp_non_finite_check_and_unscale_cuda_(Tensor& scaled_grad,
       auto* found_inf_ptr = found_inf.data_ptr<float>();
       auto* inv_scale_ptr = inv_scale.data_ptr<float>();
 
-      gpu_kernel(iter, [found_inf_ptr, inv_scale_ptr]GPU_LAMBDA(scalar_t val) -> scalar_t {
-          float fval = static_cast<float>(val);
-          // See isfinite_ensure_cuda_math above.
-          if (!isfinite_ensure_cuda_math(fval)) {
-            *found_inf_ptr = 1.f;
-          }
-          const auto inv_scale_val = *inv_scale_ptr; // Every thread accesses inv_scale, but it will hit in cache.
-          return static_cast<scalar_t>(inv_scale_val == 1.f ? fval : fval*inv_scale_val);
-        });
+      using opmath_t = get_opmath_t<scalar_t>::opmath_t;
```

Comment:
what's wrong with using acc_type<scalar_t, true>? That's what's used in all other places.

Reply:
If it does the mapping we want, that makes sense.  I'll double check the behavior.  `get_opmath_t` doesn't do anything for integer types, while `acc_type` might.  I'm not sure if we do or don't want any pre/post op casting to occur for integer types.